x-pack/filebeat/input/awss3/input.go

```diff
@@ -62,13 +63,28 @@ func (in *s3Input) Test(ctx v2.TestContext) error {
 }
 
 func (in *s3Input) Run(ctx v2.Context, pipeline beat.Pipeline) error {
-	collector, err := in.createCollector(ctx, pipeline)
-	if err != nil {
-		return err
+	var err error
+	var collector s3Collector
+
+	metricRegistry := monitoring.GetNamespace("dataset").GetRegistry()
+	inputMetrics := newInputMetrics(metricRegistry, ctx.ID)
+
+	if in.config.QueueURL != "" {
+		collector, err = in.createSQSCollector(ctx, pipeline, inputMetrics)
+		if err != nil {
+			return err
```

Comment:
maybe we should add some extra info wrap around this err to know which collector the error is from.

Reply:
👍